### PR TITLE
Enable building debian9 packages

### PR DESCRIPTION
**Pull request by rakeshsinghranchi**
This work enables building Debian 9 packages. This is the first step towards building this new distro specific package. This PR uses Debian 8.2 docker build image with static set of pre-requisite package dependency required for Debian 9.

### CODE REVIEW

**Comment by rakeshsinghranchi** [comment]
cc @Petermarcu  , @karajas . Can you take a look at the PR ? Thanks.

**Comment by Petermarcu** [comment]
@weshaggard

File: src/pkg/packaging/deb/dotnet-sharedframework-debian9_config.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    "package_name": "%SHARED_FRAMEWORK_DEBIAN_PACKAGE_NAME%",
+    "install_root": "/usr/share/dotnet",
+
+    "short_description": "%SHARED_FRAMEWORK_BRAND_NAME% %SHARED_FRAMEWORK_NUGET_NAME% %SHARED_FRAMEWORK_NUGET_VERSION%",
+    "long_description": ".NET Core is a development platform that you can use to build command-line applications, microservices and modern websites. It is open source, cross-platform and is supported by Microsoft. We hope you enjoy using it! If you do, please consider joining the active community of developers that are contributing to the project on GitHub (https://github.com/dotnet/core). We happily accept issues and PRs.",
+    "homepage": "https://dotnet.github.io",
+
+    "release":{
+        "package_version":"1.0.0.0",
```
Comment (weshaggard) [inline src/pkg/packaging/deb/dotnet-sharedframework-debian9_config.json]: Is this the correct version? Should it align with our product version?

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 2]
```diff
@@ -526,6 +526,96 @@
         "failOnStandardError": "false"
       }
     },
+	{
+      "enabled": true,
+      "continueOnError": false,
+      "alwaysRun": false,
+      "displayName": "Initialize docker - Debian 9",
```
Comment (weshaggard) [inline buildpipeline/Core-Setup-Linux-BT.json]: It looks like you are doing another full clean build for Debian 9. Is that expected? I thought we were only doing the package build but not rebuilding everything. If we plan to rebuild everything why not have a separate build leg so we can do it in parallel?

**Comment by karajas** [comment]
I believe another issue mentioned is that we don't have a cli to perform init-tools on debian 9?

File: buildpipeline/Core-Setup-Linux-BT.json (hunk 2 above)
Comment (rakeshsinghranchi) [inline buildpipeline/Core-Setup-Linux-BT.json]: No,we are not rebuilding everything . Like other native installers, we consume prebuilt Linux-x64 portable binaries by volume mapping the prebuilt binaries and build native installers for each distro using corresponding docker container.

Comment (weshaggard) [inline buildpipeline/Core-Setup-Linux-BT.json]:
```
"arguments": "run --rm $(DockerCommonRunArgs_Debian9) /bin/bash -c \"HOME=$(PB_GitDirectory); git clean -X -d -f; $(PB_GitDirectory)/init-tools.sh\"",
```

The first step you run is `git clean` which to me appears to be cleaning out the entire repo and then rebuilding. Perhaps I've missed where you are creating another clone but it certainly looks like you are using the same clone.

File: src/pkg/packaging/deb/dotnet-sharedframework-debian9_config.json (hunk 1 above)
Comment (rakeshsinghranchi) [inline src/pkg/packaging/deb/dotnet-sharedframework-debian9_config.json]: This is the default value that always gets set to the correct product version during the package build step. see https://github.com/dotnet/core-setup/blob/master/src/pkg/packaging/deb/package.targets#L178 .

File: buildpipeline/Core-Setup-Linux-BT.json (hunk 2 above)
Comment (rakeshsinghranchi) [inline buildpipeline/Core-Setup-Linux-BT.json]: Look at DockerCommonRunArgs_Debian9 build variable. It does the volume mapping to prebuilt sharedframeworkpublish folder from binary build. Also, distro specific build steps do not contain binary build step. for e.g look at https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=875149

Comment (weshaggard) [inline buildpipeline/Core-Setup-Linux-BT.json]: OK thanks for the information.

File: src/pkg/packaging/deb/dotnet-sharedframework-debian9_config.json (hunk 1 above)
Comment (weshaggard) [inline src/pkg/packaging/deb/dotnet-sharedframework-debian9_config.json]: cool.

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 3]
```diff
@@ -916,6 +1010,18 @@
     "DockerCommonRunArgs_Debian8": {
       "value": "--name $(PB_DockerContainerName)$(DockerTag_Debian8) -v \"$(PB_SourcesDirectory):$(PB_GitDirectory)\" -v $(Build.StagingDirectory)/sharedFrameworkPublish/:/root/sharedFrameworkPublish/ -w=\"$(PB_GitDirectory)\" $(DockerImageName_Debian8)"
     },
+    "DockerTag_Debian9": {
+      "value": "debian-8.2-debpkg-9f87c3c-20173003023006"
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Linux-BT.json]: @rakeshsinghranchi - why are we using a `debian-8.2` docker image to build the `debian9` deb package?

Looking at https://hub.docker.com/r/microsoft/dotnet-buildtools-prereqs/tags/, I don't see any debian 9 images.  Is this expected?  I think we should be using debian 9 machines here.

/cc @MichaelSimons

**Pull request merged**